active/InputValidation/InputValidation.md

```diff
@@ -0,0 +1,592 @@
+{
+    private string name;
+    [MinLength(4)]
+    [MaxLength(6)]
+    public string Name
+    {	
+        get { return name; }
+        set { SetValue(ref name, value); }
+    }
+}
+```
+
+#### The ViewModel
+
+The “ViewModel” property is the View Model for the Page on which this TextBox is displayed (this.ViewModel).
+``` csharp
+
+public class ViewModel
+{
+    public ViewModel()
+    {
+        Person = new Person();
+    }
+
+    public Person Person { get; set; }
+
+}
+```
+
+### [App Code] Creating a custom ValidationCommand
+Specifying a custom `InputValidationCommand` allows you to override the validation behavior and ensure that once the user has entered invalid text, validation will occur on every property change.
+
+``` csharp
+using System;
+using System.Linq;
+using Microsoft.UI.Xaml.Controls;
+using System.Reflection;
+
+namespace Microsoft.Toolkit.Uwp
+{
+    public class ValidationCommand : InputValidationCommand
+    {
+        public ValidationCommand()
+        {
+        }
+
+        public IValidatable Validator {get; set;}
+
+        protected override void ValidateCore(IInputValidationControl validationControl)
+        {
+            if (validationControl.TryGetInputPropertyValue(out object value))
+            {
+                validator.Validate(validationControl.ValidationContext.MemberName, value);
+            }
+        }
+    }
+
+    public static class IInputValidationControlExtensions
+    {
+        // Gets the value of the InputProperty declared on the control so we an properly validate.
+        public static bool TryGetInputPropertyValue(this IInputValidationControl ctrl, out object value)
+        {
+            Type controlType = ctrl.GetType();
+            value = null;
+            CustomAttributeData inputPropertyData = controlType.CustomAttributes.Where(attr => attr.AttributeType == typeof(InputPropertyAttribute)).SingleOrDefault();
+            if (inputPropertyData != null)
+            {
+                string propertyName = inputPropertyData.NamedArguments[0].TypedValue.Value.ToString();
+                value = controlType.GetProperty(propertyName).GetValue(ctrl);
+            }
+            return inputPropertyData != null;
+        }
+    }
+}
+```
+
+## Custom validation UI
+
+
+#### XAML
+A developer wants to create their own visuals for the error message style and not use the control’s visuals at all. They may want to do this to place messages far from the control or something else.
+
+To do this, you can instruct the `IInputValidationControl` to not display the visuals by setting the `InputValidationKind` property to `Hidden`.
+
+![Custom UI](customuiexample.png)
+
+```xaml
+<TextBox 
+        x:Name="Name" 
+        Header="Name: " 
+        PlaceholderText="Enter Name..." 
+
+        Text="{x:Bind   ViewModel.Person.Name, 
+                        UpdateSourceTrigger=PropertyChanged, 
+                        Mode=TwoWay}" 
+         InputValidationKind="Hidden"
```

Comment:
Will all controls that support input validation need this property added? If so it should be in the API section.